ast.hh

```diff
@@ -571,15 +587,20 @@ public:
 	/**
 	 * Bind the AST class described in the grammar to the rule specified.
 	 */
-	BindAST(const Rule &r)
+	BindAST(const Rule &r,
+	        const ErrorReporter &err = defaultErrorReporter)
 	{
-		ASTParserDelegate::bind_parse_proc(r, [](const InputRange &range,
-		                                         void *d)
+		ASTParserDelegate::bind_parse_proc(r, [&err](const InputRange &range,
```

Comment:
We do need to copy here (bind `err` to the lambda by copy, not by reference), because otherwise we end up with a pointer to a closure that may no longer be valid.